app/assets/javascripts/controllers/credentials/credentials_controller.js

```diff
@@ -44,17 +49,36 @@ ManageIQ.angular.app.controller('credentialsController', ['$http', '$scope', 'mi
     }
   };
 
+  $scope.changeStoredPrivateKey = function() {
+    $scope.bChangeStoredPrivateKey = true;
+    $scope.bCancelPrivateKeyChange = false;
+    $scope.emsCommonModel.ssh_keypair_password = '';
+  };
+
+  $scope.cancelPrivateKeyChange = function() {
+    if($scope.bChangeStoredPrivateKey) {
+      $scope.bCancelPrivateKeyChange = true;
+      $scope.bChangeStoredPrivateKey = false;
+      $scope.emsCommonModel.ssh_keypair_password = '';
```

Comment:
I recommend writing a brand new controller for this particular case, since `credentialsController` does not seem to fit the bill for the behavior that is expected here.